engines/hadesch/gfx_context.cpp

```diff
@@ -81,27 +84,63 @@ void GfxContext8Bit::fade(int val) {
 }
 
 void GfxContext8Bit::clear() {
-	memset(_pixels.get(), 0, _w * _h);
+	surf.clearPalette();
+	surf.clear();
 	memset(_palette, 0, sizeof(_palette));
 	memset(_paletteUsed, 0, sizeof(_paletteUsed));
 }
 
-GfxContext8Bit::GfxContext8Bit(int canvasW, int canvasH) {
-	_w = canvasW;
-	_h = canvasH;
-	_pitch = _w;
-	_pixels = sharedPtrByteAlloc(_w * _h);
-	memset(_palette, 0, sizeof(_palette));
-	memset(_paletteUsed, 0, sizeof(_paletteUsed));
+GfxContext8Bit::GfxContext8Bit(int canvasW, int canvasH) : surf(canvasW, canvasH, Graphics::PixelFormat::createFormatCLUT8()) {
+	clear();
 }
 
 void GfxContext8Bit::renderToScreen(Common::Point viewPoint) {
 	if (_palette) {
 		g_system->getPaletteManager()->setPalette(_palette, 0, 256);
 	}
 
-	g_system->copyRectToScreen(_pixels.get() + viewPoint.x + _pitch * viewPoint.y, _w, 0, 0,
+	g_system->copyRectToScreen(surf.getBasePtr(viewPoint.x, viewPoint.y), surf.w, 0, 0,
 				   kVideoWidth, kVideoHeight);
 }
 
+byte GfxContext8Bit::findColor(byte r, byte g, byte b) {
```

Comment:
I think we could use this opportunity to introduce something like `Graphics::Palettemanager` which will do this palette search.

I have a very similar code here:
  https://github.com/scummvm/scummvm/blob/master/graphics/macgui/macwindowmanager.cpp#L968

It is out of scope of this PR, though.